### DISCUSSION TITLE
Support for PostGIS extension

### DISCUSSION THREAD

**Discussion by tomaszburdka-tomtom** [cloudnative-pg/cloudnative-pg#215] [General]
Our projects required postgres with PostGIS extension, could you add support for that?

**Comment by sxd** [comment]
Hi Tomas!

Adding the PostGIS extension is not something that goes into CloudNativePG directly, it's something that goes in this project: https://github.com/cloudnative-pg/postgres-containers and basically it's just install PostGIS inside the container image.

If you want just to run a couple of test, you can just use the official PostGIS image here: https://hub.docker.com/r/postgis/postgis 
You just need to change the `imageName` field and point it to the tag of that image.

We're planning to work on the images and adding more extensions, but first, we need to define a base tag. Also, we're working on creating a blog post to share how to create container images compatible with the CloudNativePG project, which just need to have PostgreSQL + Barman Cloud

I'm turning this into a discussion for now if someone else wants to look into that in the future!

Regards!

**Reply by tomaszburdka-tomtom** [reply]
I've tested already some deployments with PostgresOperator using CrunchyData https://www.crunchydata.com/developers/postgresql-documentation/postgis. Is there a possibility to use image with postgis without manually changes in Docker file? In Crunchy I've just have to enable PostGis extension in some bootstrapInit sql sripts.

**Reply by gbartolini** [reply]
Just to explain though, our approach is to produce lighter base images (not everyone needs PostGIS or other extensions) and provide instructions on how to build GitHub Action based pipelines that can automatically produce such custom images starting from ours and by adding new layers.

So, basically, as long as your [images comply with these guidelines](https://cloudnative-pg.io/documentation/1.15.1/container_images/) you can run all the extensions that you want - by configuring shared_libraries correctly. In any case, we need to add the instructions for this (in the other project though, as @sxd was pointing out).

It goes without saying that we will add support for PostGIS at some stage - it's just a matter of priorities at the moment and we were busy with the open source launch and sandbox application. So any help is most welcome.

P.S.: At EDB we had developed these images for the predecessor of Cloud Native Postgres, for example: https://github.com/EnterpriseDB/docker-postgresql

**Comment by sxd** [comment]
Hi @tomaszburdka-tomtom

There's a PR created a few days a go with this support, please feel free to try the images and test it to help us merge it!
Here is the PR https://github.com/cloudnative-pg/postgres-containers/pull/22

I'm locking and closing this discussion since the implementation it's there in the PR

Regards!
